Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -73,7 +73,6 @@ public class HTTPClient {
     }
 
     deinit {
-        assert(self.pool.connectionProviderCount == 0)
```

Comment:
I wonder how important this is, we call `cancel` on all active tasks but do not wait for it to propagate to provider's `delete`...

Reply:
@artemredkin don't the providers reference the `HTTPClient`?

Reply:
No, providers only reference the pool, not the client

Reply:
Ok, I got it.